feg/gateway/services/testcore/hss/servicers/hss_s6a_proxy_integration_test.go

```diff
@@ -46,8 +47,36 @@ func TestAIR_Successful(t *testing.T) {
 	assert.Equal(t, crypto.KasmeBytes, len(vector.Kasme))
 }
 
+func TestAIR_Authentication_Rejection_WithPLMNList(t *testing.T) {
+	plmns := []string{"00101", "001022"}
+
+	s6aProxy := getTestS6aProxy(t, plmns)
+	air := &protos.AuthenticationInformationRequest{
+		UserName:                  "001010000000009",
+		VisitedPlmn:               []byte{0, 0, 0},
+		NumRequestedEutranVectors: 1,
+	}
+	// accepted IMSI
+	aia, err := s6aProxy.AuthenticationInformation(context.Background(), air)
+	assert.NoError(t, err)
+	assert.Equal(t, protos.ErrorCode_UNDEFINED, aia.ErrorCode)
+
+	assert.Equal(t, 1, len(aia.EutranVectors))
+	vector := aia.EutranVectors[0]
+	assert.Equal(t, crypto.RandChallengeBytes, len(vector.Rand))
+	assert.Equal(t, crypto.XresBytes, len(vector.Xres))
+	assert.Equal(t, crypto.AutnBytes, len(vector.Autn))
+	assert.Equal(t, crypto.KasmeBytes, len(vector.Kasme))
+
+	// rejected IMSI because does not mach any PLMN (rejection comes from S6a_proxy, not HSS)
+	air.UserName = "00102000000008"
+	aia, err = s6aProxy.AuthenticationInformation(context.Background(), air)
+	assert.NoError(t, err)
+	assert.Equal(t, protos.ErrorCode_AUTHENTICATION_REJECTED, aia.ErrorCode)
+}
```

Comment:
note this test it is kind of out of scope, since this is to test hss. But i wanted to make sure I Was not breaking anything. Also, since I don't have cwag_integ_test it helped me to test the feature. I can disable it if it's considered out of scope